src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelSourceChangeCreatedEventGenerator.java

```diff
@@ -71,15 +78,17 @@ public static EiffelSourceChangeCreatedEvent generate(PatchSetCreatedEvent patch
         eiffelEvent.eventParams.data.gitIdentifier.branch = branch;
         eiffelEvent.eventParams.data.gitIdentifier.repoName = projectName;
 
-        // TODO
-        // String latestEiffelSourceChangeCreatedEventId =
-        // getLatestEiffelSourceChangeCreatedEventId();
-        // setPreviousVersionLink(latestEiffelSourceChangeCreatedEventId);
-
-        // String latestEiffelSourceChangeSubmittedEventId =
-        // getLatestEiffelSourceChangeSubmittedEventId();
-        // setBaseLink(latestEiffelSourceChangeSubmittedEventId);
+        String previousSourceChangeCreatedEvent = getPreviousEiffelEvent(EventType.SCC_EVENT, projectName, changeId, pluginDirectoryPath);
+        final Link previousVersionLink = createLink(LINK_TYPE_PREVIOUS_VERSION, previousSourceChangeCreatedEvent);
+        if (previousVersionLink != null) {
+            eiffelEvent.eventParams.links.add(previousVersionLink);
+        }
 
+        String previousSourceChangeSubmittedEvent = getPreviousEiffelEvent(EventType.SCS_EVENT, projectName, changeId, pluginDirectoryPath);
```

Comment:
This can't be tested properly. If you try to find the SCS event based on the changeId you will always get an empty result

Reply:
As we talked about, in unittest. We won't test this since that functionality is broken apart with (this class, storage class and db handled). We will try to make sure that we are sending the correct values to each method though and later, functional tests will include this functionality.

Reply:
We have added a test to check that we are sending the correct searchCriteria.